packages/gestalt-datepicker/src/DateRange.js

```diff
@@ -0,0 +1,285 @@
+  formatRelative?: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+  localize?: {|
+    ordinalNumber: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+    era: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+    quarter: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+    month: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+    day: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+    dayPeriod: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+  |},
+  formatLong?: {|
+    date: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+    time: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+    dateTime: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+  |},
+  match?: {|
+    ordinalNumber: (...args: $ReadOnlyArray<string>) => { ... },
+    era: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+    quarter: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+    month: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+    day: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+    dayPeriod: (...args: $ReadOnlyArray<{ ... }>) => { ... },
+  |},
+  options?: {|
+    weekStartsOn?: 0 | 1 | 2 | 3 | 4 | 5 | 6,
+    firstWeekContainsDate?: 1 | 2 | 3 | 4 | 5 | 6 | 7,
+  |},
+|};
+
+type Props = {|
+  /**
+   * Customize your error message for the cases the user enters invalid end dates. See the [error messaging variant](https://gestalt.pinterest.systems/web/daterange#Error-messaging) to learn more.
+   */
+  endDateErrorMessage?: string | null,
+  /**
+   * DateRange is a controlled component. `startDateValue` sets the value of the start date.  See the [controlled component variant](https://gestalt.pinterest.systems/web/daterange#Controlled-component) to learn more.
+   */
+  endDateValue: Date | null,
+  /**
+   * DateRange accepts imported locales from the open source date utility library date-fns. See the [locales variant](https://gestalt.pinterest.systems/web/datefield#localeData) to learn more.
+   */
+  localeData?: LocaleData,
+  /**
+   * Maximal selectable date. Disables any date values after the provided date. See the [disable future and past variant](https://gestalt.pinterest.systems/web/datefield#Disable-future-and-past) to learn more.
+   */
+  maxDate?: Date,
+  /**
+   * Minimal selectable date. Disables any date values before the provided date.  See the [disable future and past variant](https://gestalt.pinterest.systems/web/datefield#Disable-future-and-past) to learn more.
+   */
+  minDate?: Date,
+  /**
+   * Callback triggered when the user clicks the Cancel button to not persist the selected dates. It should be used to close DateRange. See the [controlled component variant](https://gestalt.pinterest.systems/web/daterange#Controlled-component) to learn more.
+   */
+  onCancel: () => void,
+  /**
+   * Callback triggered when the user blurs the input of the end date DateField. See the [error messaging variant](https://gestalt.pinterest.systems/web/daterange#Error-messaging) to learn more.
```

Comment:
Maybe rephrase to: `Callback triggered when the end date input loses focus`, since I feel like a user doesn't proactively blur

Reply:
DONE